apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -230,23 +236,29 @@ def read_table(self, coords: SparseNdCoordinates) -> Iterator[pa.Table]:
                 schema=A.schema,
             )
 
-            # TODO: make a util function to be shared with DenseNdArray
-            # coords are a tuple of (int or slice-of-int)
-            if len(coords) == 1 and coords[0] == slice(None):
-                # Special case which tiledb-py supports, so we should too
-                coords = coords * A.schema.domain.ndim
-            elif len(coords) != A.schema.domain.ndim:
-                raise ValueError(
-                    f"coordinate length {len(coords)} != array ndim {A.schema.domain.ndim}"
+            # * A sequence of coordinates is accepted, one per dimension.
+            # * Sequence length must be at least one and <= number of dimensions.
+            # * If the sequence contains missing coordinates (length less than number of dimensions),
+            #   then "slice(None)" is assumed for the missing dimensions.
+            # * Per-dimension, explicitly specified coordinates can be one of: a value, a
+            #   list/ndarray/paarray/etc of values, a slice, etc.
+
+            if not (isinstance(coords, list) or isinstance(coords, tuple)):
+                # if not isinstance(coords, tuple):
+                raise SOMAError(
```

Comment:
This should be a TypeError, not a SOMAError.  See: https://docs.python.org/3/library/exceptions.html#TypeError  (last para in that section).